src/main/java/gomedic/logic/parser/editcommandparser/EditPatientCommandParser.java

```diff
@@ -0,0 +1,107 @@
+                CliSyntax.PREFIX_GENDER,
+                CliSyntax.PREFIX_HEIGHT,
+                CliSyntax.PREFIX_WEIGHT,
+                CliSyntax.PREFIX_MEDICALCONDITIONS);
+
+        Id targetId;
+
+        try {
+            targetId = ParserUtil.parseId(argMultimap.getValue(CliSyntax.PREFIX_ID).orElse(INVALID_INPUT));
+        } catch (ParseException pe) {
+            throw new ParseException(
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, EditPatientCommand.MESSAGE_USAGE), pe);
+        }
+
+        EditPatientCommand.EditPatientDescriptor editPatientDescriptor = new EditPatientCommand.EditPatientDescriptor();
+        if (argMultimap.getValue(CliSyntax.PREFIX_NAME).isPresent()) {
+            editPatientDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(CliSyntax.PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_PHONE).isPresent()) {
+            editPatientDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(CliSyntax.PREFIX_PHONE).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_AGE).isPresent()) {
+            editPatientDescriptor
+                .setAge(ParserUtil.parseAge(argMultimap.getValue(CliSyntax.PREFIX_AGE).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_BLOODTYPE).isPresent()) {
+            editPatientDescriptor
+                .setBloodType(ParserUtil.parseBloodType(argMultimap.getValue(CliSyntax.PREFIX_BLOODTYPE).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_GENDER).isPresent()) {
+            editPatientDescriptor
+                .setGender(ParserUtil.parseGender(argMultimap.getValue(CliSyntax.PREFIX_GENDER).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_HEIGHT).isPresent()) {
+            editPatientDescriptor
+                .setHeight(ParserUtil.parseHeight(argMultimap.getValue(CliSyntax.PREFIX_HEIGHT).get()));
+        }
+        if (argMultimap.getValue(CliSyntax.PREFIX_WEIGHT).isPresent()) {
+            editPatientDescriptor
+                .setWeight(ParserUtil.parseWeight(argMultimap.getValue(CliSyntax.PREFIX_WEIGHT).get()));
+        }
+        parseTagsForEdit(argMultimap.getAllValues(CliSyntax.PREFIX_MEDICALCONDITIONS))
+            .ifPresent(editPatientDescriptor::setMedicalConditions);
+
+        if (!editPatientDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditPatientCommand.MESSAGE_NOT_EDITED);
+        }
+
+        return new EditPatientCommand(targetId, editPatientDescriptor);
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
+    private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws ParseException {
+        assert tags != null;
```

Comment:
use requireNonNull here, just make is consistent

Reply:
got it!